[CI][format] Explicitly pass extensions to git-clang-format (take 2)

ldionne opened the pull request:
This patch ensures that the CI script controls which file extensions are
considered instead of letting git-clang-format apply its own filtering
rules. In particular, this properly handles libc++ extension-less headers
which were passed to git-clang-format, but then dropped by the tool as
having an unrecognized extension.

This is a second attempt to land 7620fe0d2d1e, which was reverted in 9572388 
because it caused formatting not to be enforced for several patches. The
problem was that we'd incorrectly pass the extensions with additional quoting 
to git-clang-format. The incorrect quoting has been removed in this version
of the patch.

ldionne commented:
This PR is not ready for review, I'm trying to investigate why the CI doesn't detect format breakage.

boomanaiden154 commented:
One other thing to note (if you aren't already aware) is that the python scripts are pulled in from `main` rather than the branch being merged in, so you need to modify this step if you want to test changes in the branch to the python utilities:
https://github.com/llvm/llvm-project/blob/f35318e828d166bee04a117c1a87c1c0e32a57da/.github/workflows/pr-code-format.yml#L37

ldionne commented:
@boomanaiden154 I think I found the issue. I was passing `'cpp'` as the extension. However, we're using `subprocess.run`, we're not inside a shell script. So the `'cpp'` argument was passed as-is, and `git-clang-format` would basically think there's no file with an extension of `'cpp'` (but there's a file with an extension of `cpp`).

I think removing the quoting should do it. I do need to test that the right thing happens when I have a file with no extension, especially when the only modified file is one without an extension (e.g. `libcxx/include/vector`).

ldionne committed:
[CI][format] Explicitly pass extensions to git-clang-format (take 2)

This is a second attempt to land 7620fe0, which was reverted in 9572388
because it caused formatting not to be enforced for several patches.

ldionne commented:
OK, sorry for the multiple updates here but I think this should be good to go. I tested with:

- a single `.cpp` file from LLVM
- a `.cpp` file from LLVM and an extension-less header from libc++
- a single extension-less header from libc++

and they all seemed to work now.

boomanaiden154 reviewed (approved):
LGTM.

I'll try and keep an eye on this over the next couple days after this lands to see if something drastically changes, but if the basics work, we should be good.

At some point I would like proper testing for the CI infrastructure, but haven't figured out how to do pieces of it. Hopefully I can spend some time on that soon.
